### PR TITLE
Cleanup and fixes

### PR DESCRIPTION
**Pull request by benschwartz9**
Implementing the following features:
- Integrate graphql to get correctness data
- Convert npm url to github url for calling API
- Error handling for invalid url in the file read stage (just abort the entire process)

### CODE REVIEW

File: api/api.go [hunk 1]
```diff
@@ -385,10 +410,23 @@ func NPMtoGithubUrl(npmUrl string) (githubUrl string, err error) {
...
+	//
+	// express repo available at the following URL:
+	// 	https://github.com/expressjs/express/main/
+
 	cmdOutput := string(stdout)
 	i := strings.Index(cmdOutput, "https://github.com/")
 	restOfStr := cmdOutput[i : len(cmdOutput)-1]
-	j := strings.Index(restOfStr, packageName)
+	j := strings.Index(restOfStr, "\n")     // Filter to end of new line
+	k := getNthOccurance(restOfStr, '/', 5) // Find 5th /
```
Comment (emilejbm) [inline api/api.go:422]: Why the change to use this function?

File: fileio/reader.go [hunk 2]
```diff
@@ -43,6 +43,10 @@ func ReadFile(path string, ch chan<- string) {
 		fmt.Println("fileio: read entry:", scanner.Text(), isValidURL(scanner.Text()))
 		if isValidURL(scanner.Text()) {
 			ch <- scanner.Text()
+		} else {
+			// Abort entire process if there is an invalid URL in the file
+			fmt.Fprintf(os.Stderr, "Error processing file, invalid url: %s\n", scanner.Text())
```
Comment (emilejbm) [inline fileio/reader.go:48]: Is it okay to be printing to Stderr? Otherwise, I think that this is fine

File: api/graphql.go [hunk 3]
```diff
@@ -62,14 +61,13 @@ func BuildCorrectnessQuery(ownerName string, repoName string) (query map[string]
 	return correctnessQuery
 }
 
-func GetCorrectnessFactors(ownerName string, repoName string) (watchers int64, stargazers int64, totalCommits int64, err error) {
+func GetCorrectnessFactors(url string) (watchers int64, stargazers int64, totalCommits int64, err error) {
+	ownerName, repoName, token, err := validateInput(url)
+	if err != nil {
+		return 0, 0, 0, fmt.Errorf("GetCorrectnessFactors: Error on validate input")
```
Comment (Ashassins) [inline api/graphql.go:67]: should this return -1's instead to indicate an error state?

**Review by Ashassins — APPROVED** [review]

File: api/graphql.go (hunk 3 above)
Comment (Ashassins) [reply api/graphql.go:67]: other than the faileing checks, I approve

Comment (emilejbm) [reply api/graphql.go:67]: -1 is used to say that the metric was not implemented.
